chiselFrontend/src/main/scala/chisel3/core/Assert.scala

```diff
@@ -51,12 +51,15 @@ object assert { // scalastyle:ignore object.name
   }
 
   def apply_impl_do(cond: Bool, line: String, message: Option[String], data: Bits*)(implicit sourceInfo: SourceInfo) {
+    val escLine = line.replaceAll("%", "%%")
     when (!(cond || Builder.forcedReset)) {
       val fmt = message match {
-        case Some(str) => s"Assertion failed: $str\n    at $line\n"
-        case None => s"Assertion failed\n    at $line\n"
+        case Some(str) =>
+          val msg = if (data.isEmpty) str.replaceAll("%", "%%") else str
```

Comment:
Does this make sense to have it do different things based on whether data is passed in? Can we instead make it consistent between the two cases, but error out here with an informative message if the number of format arguments doesn't match the number of data passed in?